### PR TITLE
Backport: Global Styles: Add scoping of feature level selectors

### PR DESCRIPTION
**Pull request by aaronrobertshaw**
Syncs the changes from:
- https://github.com/WordPress/gutenberg/pull/61033

These changes ensure that custom feature-level selectors for style nodes are correctly scoped when generating a theme.json stylesheet

Unit tests: `npm run test:php -- --filter Tests_Theme_wpThemeJson`

Screenshot 2024-05-02 at 12 15 45 PM

Trac ticket: https://core.trac.wordpress.org/ticket/61119

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/class-wp-theme-json.php
Comment (audrasjb) [inline src/wp-includes/class-wp-theme-json.php]: Small docblock change.

```suggestion
	/**
	 * Scopes the selectors for a given style node.
	 * 
	 * This includes the primary selector, i.e. `$node['selector']`, as well as any custom
	 * selectors for features and subfeatures, e.g. `$node['selectors']['border']` etc.
	 *
	 * @since 6.6.0
	 *
	 * @param string $scope Selector to scope to.
	 * @param array  $node  Style node with selectors to scope.
	 * @return array Node with updated selectors.
	 */
```

**Review by audrasjb — DISMISSED** [review]

File: src/wp-includes/class-wp-theme-json.php
Comment (aaronrobertshaw) [reply src/wp-includes/class-wp-theme-json.php]: Thanks for cleaning that up @audrasjb, I've committed the suggested improvement.

**Comment by vcanales** [comment]
👋 @audrasjb @aaronrobertshaw any updates here?

**Comment by aaronrobertshaw** [comment]
All the feedback to date has been addressed and we're just waiting on an approval.

**Review by tellthemachines — APPROVED** [review]
Thanks for the PR! Changes LGTM and tests are passing.

**Comment by tellthemachines** [comment]
Committed in [r58244](https://core.trac.wordpress.org/changeset/58244).

**Pull request closed without merging**
